integrations.go

```diff
@@ -374,3 +376,9 @@ func (cfi *contextifyFramesIntegration) addContextLinesToFrame(frame Frame, line
 	}
 	return frame
 }
+
+func (cfi *contextifyFramesIntegration) updateCachedLocation(key, val string) {
+	cfi.mu.Lock()
```

Comment:
This is a write-lock. I can't spot a use of a read-lock. Is the reader-write really needed here as opposed to a plain mutex?